animation_nodes/ui/data_input_panel.py

```diff
@@ -13,18 +13,21 @@ class ViewportInputPanel(bpy.types.Panel):
     def draw(self, context):
         layout = self.layout
         nodes = getNodesByType("an_ViewportInputNode")
-        for node in nodes:
-            box = layout.box()
+        if len(nodes) == 0:
```

Comment:
Can you use a guard condition instead?

Reply:
you mean like 
```
if len(nodes) == 0:
    layout.label(text="No Viewport Input node", icon="INFO")
    return
for node in nodes:
    box = layout.box()
```

Reply:
done ! not a really good coder here, i'm not used to this syntax, is it just a readability issue, or does it improve code performance ?

Reply:
It probably doesn't affect performance, but it improves readability.